allure-app-under-test/src/test/java/com/capgemini/ntc/security/auth/AuthRequiredTest.java

```diff
@@ -0,0 +1,88 @@
+package com.capgemini.ntc.security.auth;
+
+import static io.restassured.RestAssured.given;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.apache.http.HttpStatus;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import com.capgemini.ntc.security.EnvironmentParam;
+import com.capgemini.ntc.security.SecurityTest;
+import com.capgemini.ntc.security.SubUrlEnum;
+import com.capgemini.ntc.security.session.SessionEnum;
+
+import io.restassured.http.Method;
+import io.restassured.specification.RequestSpecification;
+
+/**
+ * The test verifies, that protected resources can not be accessed by unauthenticated
+ * users.
+ * OWASP ASVS requirement V2.1: Verify all pages and resources by default
+ * require authentication except those specifically intended to be public
+ * (Principle of complete mediation)
+ * Purpose: This is one of the most basic protection mechanisms. Unit test
+ * verifies only a part of this requirement (resources that require authentication -
+ * enforce authentication). The second part of this requirement ("by default
+ * require") must be validated manually.
+ * Read also:
+ * [1]
+ * https://www.owasp.org/index.php/Authentication_Cheat_Sheet
+ *
+ * @author Marek Puchalski, Capgemini
+ */
+@RunWith(Parameterized.class)
```

Comment:
As library `org.junit.runners.Parameterized` has many pitfalls and limitations we use  `@RunWith(JUnitParamsRunner.class)`.  

Here is code example:
https://github.com/devonfw/devonfw-testing/blob/develop/allure-app-under-test/src/test/java/com/capgemini/ntc/core/datadriven/DataDrivenInternalExampleTest.java

Here are links how to use it: 
https://dzone.com/articles/working-with-junitparams
https://github.com/Pragmatists/JUnitParams